internal/app/document_store.go

```diff
@@ -0,0 +1,109 @@
+package app
+
+import (
+	"context"
+	"errors"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/dynamo"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/page"
+)
+
+type documentStore struct {
+	dynamoClient DynamoClient
+	s3Client     S3Client
+}
+
+func NewDocumentStore(dynamoClient DynamoClient, s3Client S3Client) DocumentStore {
+	return documentStore{dynamoClient: dynamoClient, s3Client: s3Client}
```

Comment:
probably makes sense to have this return `&documentStore` and change all the receivers to `(s *documentStore)`